# LISTA 11 - PANDAS

## <ins>Ejercicios Obligatorios </ins>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pandas import Series, DataFrame

## EJERCICIO 1

Crea un dataframe con 15 filas y 2 columnas. La primera columna se llamará x, la segunda y. Cada entrada será un número aleatorio (puedes usar random.normal() de la librería numpy). Guarda el dataframe en una variable llamada points.

In [2]:
import pandas as pd
import numpy as np
data = np.random.normal(size=(15, 2)) # Crea una sample de datos aleatorios de 15 filas en 2 columnas
points = pd.DataFrame(data, columns=['x', 'y']) # Crea un dataframe con la matrix de 15x2 y le pone de cabecera la lista de columns
print(points)

           x         y
0   0.070148  0.407486
1  -0.781381 -0.245572
2   0.894044  1.409231
3   0.616098  1.064374
4  -0.141509  0.741211
5  -1.031070 -1.224261
6   1.384198  1.451956
7   2.351305 -0.116768
8  -0.662139 -0.711534
9   0.397724  0.533483
10  1.785779  0.105086
11  1.406627 -0.922734
12 -0.340750  0.789104
13 -0.416812 -0.307955
14 -0.230535  0.205743


## EJERCICIO 2

Del dataframe points, muestra las filas cuyo valor en la columna x sea positivo.

In [3]:
print(points[points['x']>0]) # points['x']>0 genera una columna de booleanos que si se usa de indice indica si imprimir o no la fila

           x         y
0   0.070148  0.407486
2   0.894044  1.409231
3   0.616098  1.064374
6   1.384198  1.451956
7   2.351305 -0.116768
9   0.397724  0.533483
10  1.785779  0.105086
11  1.406627 -0.922734


## EJERCICIO 3

In [4]:
print(points[points['y']<0])

           x         y
1  -0.781381 -0.245572
5  -1.031070 -1.224261
7   2.351305 -0.116768
8  -0.662139 -0.711534
11  1.406627 -0.922734
13 -0.416812 -0.307955


Del dataframe points, muestra las filas cuyo valor en la columna y sea negativo.

## EJERCICIO 4

In [5]:
print(points[(points['x'] > 0) & (points['y'] > 0)]) # Hace un AND entre las columnas booleanas de x e y para sacar las que ambas sean positivas

           x         y
0   0.070148  0.407486
2   0.894044  1.409231
3   0.616098  1.064374
6   1.384198  1.451956
9   0.397724  0.533483
10  1.785779  0.105086


Del dataframe points, muestra las observaciones cuyos puntos (x, y) pertenezcan al primer cuadrante. 

## EJERCICIO 5

In [6]:
for i,row in points.iterrows():
    sino = "si" if (row['x'] > 0) and (row['y'] > 0) else "no"
    print(f"{i} El punto ({row['x']:.4f},{row['y']:.4f}) {sino} pertenece al primer cuadrante") # ..4f limita a 4 decimales y redondea

0 El punto (0.0701,0.4075) si pertenece al primer cuadrante
1 El punto (-0.7814,-0.2456) no pertenece al primer cuadrante
2 El punto (0.8940,1.4092) si pertenece al primer cuadrante
3 El punto (0.6161,1.0644) si pertenece al primer cuadrante
4 El punto (-0.1415,0.7412) no pertenece al primer cuadrante
5 El punto (-1.0311,-1.2243) no pertenece al primer cuadrante
6 El punto (1.3842,1.4520) si pertenece al primer cuadrante
7 El punto (2.3513,-0.1168) no pertenece al primer cuadrante
8 El punto (-0.6621,-0.7115) no pertenece al primer cuadrante
9 El punto (0.3977,0.5335) si pertenece al primer cuadrante
10 El punto (1.7858,0.1051) si pertenece al primer cuadrante
11 El punto (1.4066,-0.9227) no pertenece al primer cuadrante
12 El punto (-0.3408,0.7891) no pertenece al primer cuadrante
13 El punto (-0.4168,-0.3080) no pertenece al primer cuadrante
14 El punto (-0.2305,0.2057) no pertenece al primer cuadrante


Del dataframe points, muestra las observaciones de la forma: “El punto ({x}, {y}) {no/sí} pertenece al primer cuadrante”.

## EJERCICIO 6

In [7]:
header = ['word']
word = ["euro", "diez", "algas", "broma", "cicuta", "fatiga", "nachos", "jadeos", "hazañas", "boutique"]
words = pd.DataFrame(data=word,columns=header)
words['length'] = words['word'].apply(len) # Usa la función por cada item de cada fila para generar la columna

def countVocals(word):
    n = 0
    for letter in word:
        if letter.lower() in 'aeiouáéíóú':
            n += 1
    return n
    
words['vowels'] = words['word'].apply(countVocals)
words['consonants'] = words['length'] - words['vowels']
print(words)

       word  length  vowels  consonants
0      euro       4       3           1
1      diez       4       2           2
2     algas       5       2           3
3     broma       5       2           3
4    cicuta       6       3           3
5    fatiga       6       3           3
6    nachos       6       2           4
7    jadeos       6       3           3
8   hazañas       7       3           4
9  boutique       8       5           3


Crea un dataframe con 10 filas y 4 columnas. La primera columna se llamará word; la segunda, length; la tercera, vowels; y la última, consonants. La columna words contendrá las siguientes 10 palabras: “euro”, “diez” “algas”, “broma”, “cicuta”, “fatiga”, “nachos”, “jadeos”, “hazañas”, “boutique”. Las columnas length, vowels y consonants contendrán, respectivamente, la longitud, el total de vocales y el total de consonantes (añade las columnas calculando su valor por código). Guarda el dataframe en la variable words.

## EJERCICIO 7

In [8]:
for i,row in words.iterrows():
    print(f"{i} La palabra {row['word']} tiene {row['length']} letras, de las cuales {row['vowels']} son vocales y {row['consonants']} son consonantes.")

0 La palabra euro tiene 4 letras, de las cuales 3 son vocales y 1 son consonantes.
1 La palabra diez tiene 4 letras, de las cuales 2 son vocales y 2 son consonantes.
2 La palabra algas tiene 5 letras, de las cuales 2 son vocales y 3 son consonantes.
3 La palabra broma tiene 5 letras, de las cuales 2 son vocales y 3 son consonantes.
4 La palabra cicuta tiene 6 letras, de las cuales 3 son vocales y 3 son consonantes.
5 La palabra fatiga tiene 6 letras, de las cuales 3 son vocales y 3 son consonantes.
6 La palabra nachos tiene 6 letras, de las cuales 2 son vocales y 4 son consonantes.
7 La palabra jadeos tiene 6 letras, de las cuales 3 son vocales y 3 son consonantes.
8 La palabra hazañas tiene 7 letras, de las cuales 3 son vocales y 4 son consonantes.
9 La palabra boutique tiene 8 letras, de las cuales 5 son vocales y 3 son consonantes.


Muestra las 10 observaciones de words con el formato “La palabra {word} tiene {length} letras, de las cuales {vowels} son vocales y {consonants} consonantes”.

## EJERCICIO 8

In [9]:
for i,row in words.iterrows():
    if row['vowels'] == row['consonants']:
        print(f"La palabra {row['word']} tiene la misma cantidad de vocales y consonantes.")

La palabra diez tiene la misma cantidad de vocales y consonantes.
La palabra cicuta tiene la misma cantidad de vocales y consonantes.
La palabra fatiga tiene la misma cantidad de vocales y consonantes.
La palabra jadeos tiene la misma cantidad de vocales y consonantes.


Muestra aquellas observaciones de words que tienen el mismo número de vocales y consonantes.

## EJERCICIO 9

In [10]:
fotocasa_df = pd.read_csv("https://drive.google.com/uc?export=view&id=1DwvALe87aeDzTk7s6xGPDd6itJ65s4YE", index_col = 0)
fotocasa_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 617 entries, 0 to 616
Data columns (total 59 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Precio               598 non-null    object
 1   Título               616 non-null    object
 2   Descripción          588 non-null    object
 3   Fotos                582 non-null    object
 4   Dirección            597 non-null    object
 5   Inmobiliaria         617 non-null    bool  
 6   Anunciante           598 non-null    object
 7   Habitaciones         551 non-null    object
 8   Baños                592 non-null    object
 9   Superficie           586 non-null    object
 10  Tipo de inmueble     597 non-null    object
 11  Ascensor             494 non-null    object
 12  Amueblado            469 non-null    object
 13  Depósito             136 non-null    object
 14  Gastos de comunidad  390 non-null    object
 15  Consumo energía      598 non-null    object
 16  Emisiones    

El siguiente DataFrame contiene información de los pisos en alquiler en Murcia a fecha de 31/12/2020:
```
fotocasa_df = pd.read_csv("https://drive.google.com/uc?export=view&id=1DwvALe87aeDzTk7s6xGPDd6itJ65s4YE", index_col = 0)
```
Muestra el número de filas, columnas y valores que contiene el DataFrame cargado:

## EJERCICIO 10

In [11]:
fotocasa_df.head()

Precio                                     Título  \
0  A consultar                 Piso de alquiler en Centro   
1   500 € /mes          Apartamento de alquiler en Centro   
2   900 € /mes  Piso de alquiler en Doctor Marañón Centro   
3   750 € /mes                 Piso de alquiler en Centro   
4   800 € /mes                 Piso de alquiler en Centro   

                                         Descripción     Fotos  \
0  TRABAJADORES Y ESTUDIANTES ATENCION!!!! VIVIEN...  18 Fotos   
1  Maravilloso apartamento para entrar a vivir en...  21 Fotos   
2  Se alquila estupenda vivienda recién reformada...  18 Fotos   
3  Te presentamos este fabuloso apartamento situa...  30 Fotos   
4  Fantástica vivienda en alquiler ubicada muy pr...  28 Fotos   

               Dirección  Inmobiliaria                    Anunciante  \
0                 Centro          True      DPTO. COMERCIAL PISOSMUR   
1                 Centro          True                BREXTOM MURCIA   
2  Doctor Marañón Centro          True  ABAD SERVICIOS INMOBILIARIOS   
3                 Centro          True  ABAD SERVICIOS INMOBILIARIOS   
4                 Centro          True                BREXTOM MURCIA   

                        Habitaciones                              Baños  \
0  <span><span>4</span> habs.</span>  <span><span>2</span> baños</span>   
1   <span><span>1</span> hab.</span>   <span><span>1</span> baño</span>   
2  <span><span>4</span> habs.</span>  <span><span>2</span> baños</span>   
3  <span><span>2</span> habs.</span>   <span><span>1</span> baño</span>   
4  <span><span>3</span> habs.</span>  <span><span>2</span> baños</span>   

                         Superficie  ... Serv. portería Internet  \
0  <span><span>110</span> m²</span>  ...            NaN      NaN   
1   <span><span>46</span> m²</span>  ...            NaN      NaN   
2  <span><span>169</span> m²</span>  ...            NaN      NaN   
3   <span><span>77</span> m²</span>  ...            NaN      NaN   
4  <span><span>104</span> m²</span>  ...            NaN      NaN   

  Suite - con baño Patio Domótica Pista de Tenis Alarma Mascotas  \
0              NaN   NaN      NaN            NaN    NaN      NaN   
1              NaN   NaN      NaN            NaN    NaN      NaN   
2              NaN   NaN      NaN            NaN    NaN      NaN   
3              NaN   NaN      NaN            NaN    NaN      NaN   
4              NaN   NaN      NaN            NaN    NaN      NaN   

  Energía Solar Gimnasio  
0           NaN      NaN  
1           NaN      NaN  
2           NaN      NaN  
3           NaN      NaN  
4           NaN      NaN  

[5 rows x 59 columns]

Recupera los cinco primeros pisos (filas)

## EJERCICIO 11

In [12]:
fotocasa_df.tail()

Precio                                             Título Descripción  \
612    NaN  Alquiler piso en barrio dEl Carmen en El Carme...         NaN   
613    NaN  Alquiler apartamento en plaza san nicolas en M...         NaN   
614    NaN  Alquiler apartamento exclusivo apto plaza flor...         NaN   
615    NaN  Alquiler piso alquiler de local entresuelo en ...         NaN   
616    NaN  Alquiler piso en alquiler en edificio alba, sa...         NaN   

    Fotos Dirección  Inmobiliaria Anunciante Habitaciones Baños Superficie  \
612   NaN       NaN          True        NaN          NaN   NaN        NaN   
613   NaN       NaN          True        NaN          NaN   NaN        NaN   
614   NaN       NaN          True        NaN          NaN   NaN        NaN   
615   NaN       NaN          True        NaN          NaN   NaN        NaN   
616   NaN       NaN          True        NaN          NaN   NaN        NaN   

     ... Serv. portería Internet Suite - con baño Patio Domótica  \
612  ...            NaN      NaN              NaN   NaN      NaN   
613  ...            NaN      NaN              NaN   NaN      NaN   
614  ...            NaN      NaN              NaN   NaN      NaN   
615  ...            NaN      NaN              NaN   NaN      NaN   
616  ...            NaN      NaN              NaN   NaN      NaN   

    Pista de Tenis Alarma Mascotas Energía Solar Gimnasio  
612            NaN    NaN      NaN           NaN      NaN  
613            NaN    NaN      NaN           NaN      NaN  
614            NaN    NaN      NaN           NaN      NaN  
615            NaN    NaN      NaN           NaN      NaN  
616            NaN    NaN      NaN           NaN      NaN  

[5 rows x 59 columns]

Recupera los cinco últimos pisos:

Muestra los campos Inmobiliaria y Tipo de inmueble de los tres últimos pisos:

## EJERCICIO 12

In [13]:
fotocasa1_df = fotocasa_df.drop(labels = ["Domótica", "Pista de Tenis","Alarma","Mascotas","Energía Solar","Gimnasio"], axis = 1)
fotocasa1_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 617 entries, 0 to 616
Data columns (total 53 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Precio               598 non-null    object
 1   Título               616 non-null    object
 2   Descripción          588 non-null    object
 3   Fotos                582 non-null    object
 4   Dirección            597 non-null    object
 5   Inmobiliaria         617 non-null    bool  
 6   Anunciante           598 non-null    object
 7   Habitaciones         551 non-null    object
 8   Baños                592 non-null    object
 9   Superficie           586 non-null    object
 10  Tipo de inmueble     597 non-null    object
 11  Ascensor             494 non-null    object
 12  Amueblado            469 non-null    object
 13  Depósito             136 non-null    object
 14  Gastos de comunidad  390 non-null    object
 15  Consumo energía      598 non-null    object
 16  Emisiones    

Las columnas Domótica, Pista de Tenis, Alarma, Mascotas, Energía Solar y Gimnasio no contienen apenas valores definidos. Elimínalas del DataFrame y almacena el resultado en fotocasa1_df:

## EJERCICIO 13

In [14]:
fotocasa2_df = fotocasa1_df.rename(columns={'Tipo de inmueble': 'Tipo'})
print(fotocasa2_df['Tipo'].head())

0           Piso
1    Apartamento
2           Piso
3           Piso
4           Piso
Name: Tipo, dtype: object


Renombra a continuación la columna Tipo de inmueble de fotocasa1_df por Tipo. Asigna el resultado a la variable fotocasa2_df:

## EJERCICIO 14

In [15]:
fotocasa4_df = fotocasa2_df[fotocasa2_df['Precio'].notnull() & (fotocasa2_df['Precio'] != "A consultar")]
print(fotocasa2_df['Precio'])
print("\n\n")
fotocasa4_df['Precio']

0      A consultar
1       500 € /mes
2       900 € /mes
3       750 € /mes
4       800 € /mes
          ...     
612            NaN
613            NaN
614            NaN
615            NaN
616            NaN
Name: Precio, Length: 617, dtype: object





1      500 € /mes
2      900 € /mes
3      750 € /mes
4      800 € /mes
5      650 € /mes
          ...    
589    600 € /mes
590    800 € /mes
591    800 € /mes
593    800 € /mes
596    850 € /mes
Name: Precio, Length: 588, dtype: object

Elimina del DataFrame aquellos pisos que no tienen Precio o que su Precio sea "A consultar". Asigna el resultado a la variable fotocasa4_df:

PISTA: para seleccionar los alquileres sin precio utiliza el método isnull(). Utiliza notnull() para seleccionar los alquileres con precio..

## EJERCICIO 15

In [16]:
convert = ["Tipo","Orientación","Estado","Parking"]
fotocasa5_df = fotocasa4_df.copy()
fotocasa5_df[convert] = fotocasa4_df[convert].astype("category")
fotocasa5_df[convert].dtypes

Tipo           category
Orientación    category
Estado         category
Parking        category
dtype: object

Utiliza la función astype para convertir las columnas Tipo, Orientación, Estado y Parking a tipo categórico. Asigna los resultados a la variable fotocasa5_df y comprueba mediante el parámetro dtype que el tipo asignado sea correcto:

## EJERCICIO 16

Cargar el fichero situado en https://raw.githubusercontent.com/ainaramu-icjardin/big_data/main/ciudades_ejemplo.csv
como un dataframe Pandas, y guardarlo a continuación en formato excel.

## EJERCICIO 17

Cargar el fichero "https://raw.githubusercontent.com/ainaramu-icjardin/big_data/main/parocomunidades.csv" que está con codificación (encoding) "latin1" y dejarlo en un dataframe df_paro.

Mostrar solo las filas de `df_paro` que corresponden al Periodo 2019.

Mostrar solo las filas de df_paro que corresponden a un total mayor de 15.

Mostrar solo las filas que corresponden al Periodo 2019 y tienen Total mayor de 15.

## <ins>Ejercicios Opcionales </ins>

## EJERCICIO 18

In [17]:
words.sort_values(by="length",ascending=False)

word  length  vowels  consonants
9  boutique       8       5           3
8   hazañas       7       3           4
4    cicuta       6       3           3
5    fatiga       6       3           3
6    nachos       6       2           4
7    jadeos       6       3           3
2     algas       5       2           3
3     broma       5       2           3
0      euro       4       3           1
1      diez       4       2           2

Investiga el método .sort_values() para ordenar las observaciones según la longitud de las palabras en orden descendente.

## EJERCICIO 19

In [18]:
l_vowels = sorted(words['vowels'].tolist(),reverse=True)
print(l_vowels)

[5, 3, 3, 3, 3, 3, 2, 2, 2, 2]


Convierte la columna vowels a lista y, con sorted() ordénala de mayor a menor. Investiga para ello el método .tolist().

## EJERCICIO 20

In [19]:
fotocasa100_df = fotocasa5_df[fotocasa5_df.index % 100 == 0]
fotocasa100_df

Precio                                             Título  \
100  540 € /mes  Piso de alquiler en C/ José Hernández - Ed Alc...   
200  700 € /mes                  Apartamento de alquiler en Centro   
300  850 € /mes      Piso de alquiler en Juan de Borbón - La Flota   
400  700 € /mes                         Piso de alquiler en Centro   
500  550 € /mes  Piso de alquiler en Calle del Platanito Vistab...   

                                           Descripción     Fotos  \
100  PROMOCIÓN: 1 MES DE ALQUILER GRATIS si reserva...  27 Fotos   
200  Alquiler de apartamento en el centro, junto a ...  19 Fotos   
300  Piso completamente amueblado y cocina equipada...  13 Fotos   
400  Piso completamente amueblado y cocina equipada...  10 Fotos   
500  Vivienda 4 dormitorios en Avenida de la Fama! ...   9 Fotos   

                                           Dirección  Inmobiliaria  \
100  C/ José Hernández - Ed Alcántara II - Espinardo          True   
200                                           Centro          True   
300                        Juan de Borbón - La Flota          True   
400                                           Centro          True   
500                   Calle del Platanito Vistabella          True   

                             Anunciante                       Habitaciones  \
100  SOLVIA SERVICIOS INMOBILIARIOS S.L  <span><span>2</span> habs.</span>   
200              BASCUÑANA INMOBILIARIA  <span><span>2</span> habs.</span>   
300   THE SIMPLE RENT LUXURY PROPERTIES  <span><span>3</span> habs.</span>   
400                         OPAU MURCIA  <span><span>2</span> habs.</span>   
500                            LOGAR 21  <span><span>4</span> habs.</span>   

                                 Baños                       Superficie  ...  \
100   <span><span>1</span> baño</span>  <span><span>82</span> m²</span>  ...   
200  <span><span>2</span> baños</span>  <span><span>70</span> m²</span>  ...   
300  <span><span>2</span> baños</span>  <span><span>90</span> m²</span>  ...   
400  <span><span>2</span> baños</span>  <span><span>45</span> m²</span>  ...   
500   <span><span>1</span> baño</span>  <span><span>95</span> m²</span>  ...   

    Z. Comunitaria Piscina   TV Zona Deportiva Zona Infantil  \
100            NaN     NaN  NaN            NaN           NaN   
200            NaN     NaN  NaN            NaN           NaN   
300            NaN     NaN  NaN            NaN           NaN   
400            NaN     NaN  NaN            NaN           NaN   
500           True     NaN  NaN            NaN           NaN   

    Piscina comunitaria Serv. portería Internet Suite - con baño Patio  
100                 NaN            NaN      NaN              NaN   NaN  
200                 NaN            NaN      NaN              NaN   NaN  
300                 NaN            NaN      NaN             True   NaN  
400                 NaN            NaN      NaN              NaN   NaN  
500                 NaN            NaN      NaN              NaN   NaN  

[5 rows x 53 columns]

Recupera aquellos pisos que ocupan una posición múltiplo de 100:

## EJERCICIO 21

In [20]:
fotocasa5_df.iloc[7]

Precio                                                        450 € /mes
Título                                     Piso de alquiler en El Carmen
Descripción            Piso de 2 dormitorios equipado completo, cocin...
Fotos                                                            6 Fotos
Dirección                                                      El Carmen
Inmobiliaria                                                        True
Anunciante                                                   OPAU MURCIA
Habitaciones                           <span><span>2</span> habs.</span>
Baños                                   <span><span>1</span> baño</span>
Superficie                               <span><span>85</span> m²</span>
Tipo                                                                Piso
Ascensor                                                             NaN
Amueblado                                                             Sí
Depósito                                           

Muestra el piso que se encuentra en la posición 7:

## EJERCICIO 22

In [21]:
fotocasa5_df.loc[9]

Precio                                                      1.400 € /mes
Título                                        Piso de alquiler en Centro
Descripción            Piso de 3 dormitorios, equipados completos, la...
Fotos                                                           16 Fotos
Dirección                                                         Centro
Inmobiliaria                                                        True
Anunciante                             THE SIMPLE RENT LUXURY PROPERTIES
Habitaciones                           <span><span>3</span> habs.</span>
Baños                                  <span><span>2</span> baños</span>
Superficie                              <span><span>150</span> m²</span>
Tipo                                                                Piso
Ascensor                                                              Sí
Amueblado                                                             Sí
Depósito                                           

Muestra el piso cuyo índice vale 9:

## EJERCICIO 23

In [22]:
print(fotocasa5_df['Dirección'].head(3))
print("\n\n")
print(fotocasa5_df[['Dirección']].head(3))

1                   Centro
2    Doctor Marañón Centro
3                   Centro
Name: Dirección, dtype: object



               Dirección
1                 Centro
2  Doctor Marañón Centro
3                 Centro


Muestra el campo Dirección de los tres primeros pisos. Utiliza para ello tanto el operador ```[]``` como ```[[]]```. ¿Qué diferencia encuentras?

Que en el segundo se omite la información sobre la columna

## EJERCICIO 24

In [25]:
fotocasa5_df[["Tipo","Inmobiliaria"]].tail(3)

Tipo  Inmobiliaria
591  Piso          True
593  Piso          True
596  Piso          True

Muestra los campos Inmobiliaria y Tipo de inmueble de los tres últimos pisos:

## EJERCICIO 25

In [27]:
fotocasa5_df.iloc[2,2]

'Te presentamos este fabuloso apartamento situado en calle Balsas de Murcia, junto a Calle Correos. Zona céntrica con todos los servicios y a un minuto de la Catedral y la Universidad de la Merced. Se trata de una vivienda de dos dormitorios grandes, cocina independiente comunicada con el salón por un cristal, baño completo, recibidor y salón con balcón. La vivienda está reformada, para entrar a vivir con tarima, pintura lisa, puertas de madera lacadas en blanco, cocina equipada y tres consolas de aire acondicionado. Se alquila con muebles. Además, cuenta con la mejor orientación Levante mediodía. Llámanos y visítala!!'

Muestra el valor del DataFrame contenido en la posición (2,2):

## EJERCICIO 26

In [28]:
fotocasa5_df.loc[2,"Título"]

'Piso de alquiler en Doctor Marañón Centro'

Muestra el valor del DataFrame cuyo índice es 2 y cuya columna es Título:

## EJERCICIO 27

In [29]:
fotocasa5_df[fotocasa5_df["Tipo"] == "Apartamento"].head(3)

Precio                                Título  \
1   500 € /mes     Apartamento de alquiler en Centro   
12  500 € /mes  Apartamento de alquiler en Espinardo   
15  400 € /mes  Apartamento de alquiler en El Carmen   

                                          Descripción     Fotos  Dirección  \
1   Maravilloso apartamento para entrar a vivir en...  21 Fotos     Centro   
12  Apartamento de 1 dormitorio,totalmente amuebla...   8 Fotos  Espinardo   
15  Apartamento con 1 dormitorio independiente, eq...   5 Fotos  El Carmen   

    Inmobiliaria      Anunciante                      Habitaciones  \
1           True  BREXTOM MURCIA  <span><span>1</span> hab.</span>   
12          True     OPAU MURCIA  <span><span>1</span> hab.</span>   
15          True     OPAU MURCIA  <span><span>1</span> hab.</span>   

                               Baños                       Superficie  ...  \
1   <span><span>1</span> baño</span>  <span><span>46</span> m²</span>  ...   
12  <span><span>1</span> baño</span>  <span><span>40</span> m²</span>  ...   
15  <span><span>1</span> baño</span>  <span><span>33</span> m²</span>  ...   

   Z. Comunitaria Piscina   TV Zona Deportiva Zona Infantil  \
1             NaN     NaN  NaN            NaN           NaN   
12            NaN     NaN  NaN            NaN           NaN   
15            NaN     NaN  NaN            NaN           NaN   

   Piscina comunitaria Serv. portería Internet Suite - con baño Patio  
1                  NaN            NaN      NaN              NaN   NaN  
12                 NaN            NaN      NaN              NaN   NaN  
15                 NaN            NaN      NaN              NaN   NaN  

[3 rows x 53 columns]

Muestra los tres primeros inmuebles de Tipo de inmueble = Apartamento:

## EJERCICIO 28

In [31]:
fotocasa5_df[(fotocasa5_df["Tipo"] == "Piso") & (fotocasa5_df["Inmobiliaria"] == False)].head(2)

Precio                                             Título  \
36  550 € /mes  Piso de alquiler en Calle Joaquín Báguena, 10 ...   
38  560 € /mes  Piso de alquiler en Avenida del Palmar, 44 Ron...   

                                          Descripción     Fotos  \
36  Alquilo piso centrico y muy cuidado, con parqu...  15 Fotos   
38  RONDA SUR (Zona innovadora)\nParticular (Sin c...  20 Fotos   

                           Dirección  Inmobiliaria  \
36  Calle Joaquín Báguena, 10 Centro         False   
38  Avenida del Palmar, 44 Ronda Sur         False   

                                           Anunciante  \
36  <span class="re-ContactDetail-particularLabel"...   
38  <span class="re-ContactDetail-particularLabel"...   

                         Habitaciones                             Baños  \
36  <span><span>2</span> habs.</span>  <span><span>1</span> baño</span>   
38  <span><span>2</span> habs.</span>  <span><span>1</span> baño</span>   

                         Superficie  ... Z. Comunitaria Piscina    TV  \
36  <span><span>53</span> m²</span>  ...            NaN     NaN   NaN   
38  <span><span>78</span> m²</span>  ...            NaN     NaN  True   

   Zona Deportiva Zona Infantil Piscina comunitaria Serv. portería Internet  \
36            NaN           NaN                 NaN            NaN      NaN   
38            NaN           NaN                 NaN            NaN      NaN   

   Suite - con baño Patio  
36              NaN   NaN  
38              NaN   NaN  

[2 rows x 53 columns]

Muestra los dos primeros inmuebles de Tipo de inmueble = Piso y que no se alquilen por inmobiliaria (Inmobiliaria = False):

## EJERCICIO 29

In [32]:
fotocasa3_df = fotocasa2_df.iloc[597:599]
fotocasa2_df.drop(fotocasa2_df.index[597:599])
fotocasa3_df

Precio                                      Título  \
597  A consultar  Piso de alquiler en San Andrés - San Antón   
598  A consultar               Piso de alquiler en El Ranero   

                                           Descripción     Fotos  \
597  ATENCIÓN ESTUDIANTES, MAGNÍFICA VIVIENDA RONDA...  10 Fotos   
598  ATENCIÓN ESTUDIANTES JUNTO A LA ESCUELA DE IDI...   7 Fotos   

                  Dirección  Inmobiliaria                Anunciante  \
597  San Andrés - San Antón          True  DPTO. COMERCIAL PISOSMUR   
598               El Ranero          True  DPTO. COMERCIAL PISOSMUR   

                          Habitaciones                              Baños  \
597  <span><span>4</span> habs.</span>  <span><span>2</span> baños</span>   
598  <span><span>4</span> habs.</span>  <span><span>2</span> baños</span>   

                           Superficie  ... Z. Comunitaria Piscina    TV  \
597  <span><span>120</span> m²</span>  ...            NaN    True   NaN   
598                               NaN  ...            NaN     NaN  True   

    Zona Deportiva Zona Infantil Piscina comunitaria Serv. portería Internet  \
597            NaN           NaN                 NaN            NaN      NaN   
598            NaN           NaN                 NaN            NaN      NaN   

    Suite - con baño Patio  
597              NaN   NaN  
598              NaN   NaN  

[2 rows x 53 columns]

Elimina del DataFrame contenido en fotocasa2_df los alquileres de la posición 597 y 598. Asigna el resultado a fotocasa3_df: